data/transactions/logic/evalStateful_test.go

```diff
@@ -324,27 +324,35 @@ log
 	}
 
 	// check stateful opcodes are not allowed in stateless mode
-	statefulOpcodeCalls := []string{
-		"txn Sender; balance",
-		"txn Sender; min_balance",
-		"txn Sender; int 0; app_opted_in",
-		"txn Sender; int 0; byte 0x01; app_local_get_ex",
-		"byte 0x01; app_global_get",
-		"int 0; byte 0x01; app_global_get_ex",
-		"txn Sender; byte 0x01; byte 0x01; app_local_put",
-		"byte 0x01; int 0; app_global_put",
-		"txn Sender; byte 0x01; app_local_del",
-		"byte 0x01; app_global_del",
-		"txn Sender; int 0; asset_holding_get AssetFrozen",
-		"int 0; int 0; asset_params_get AssetManager",
-		"int 0; int 0; app_params_get AppApprovalProgram",
-		"byte 0x01; log",
-	}
-
-	for _, source := range statefulOpcodeCalls {
-		source := source
-		testLogic(t, source, AssemblerMaxVersion, defaultEvalParams(),
-			"not allowed in current mode", "not allowed in current mode")
+	for v := uint64(2); v <= AssemblerMaxVersion; v++ {
+		sender := "txn Sender;"
+		if v < directRefEnabledVersion {
+			sender = "int 0;"
+		}
+		statefulOpcodeCalls := map[string]uint64{
+			sender + "balance":                                2,
+			sender + "min_balance":                            3,
+			sender + "int 0; app_opted_in":                    2,
+			sender + "int 0; byte 0x01; app_local_get_ex":     2,
+			"byte 0x01; app_global_get":                       2,
+			"int 0; byte 0x01; app_global_get_ex":             2,
+			sender + "byte 0x01; byte 0x01; app_local_put":    2,
+			"byte 0x01; int 0; app_global_put":                2,
+			sender + "byte 0x01; app_local_del":               2,
+			"byte 0x01; app_global_del":                       2,
+			sender + "int 0; asset_holding_get AssetFrozen":   2,
+			"int 0; int 0; asset_params_get AssetManager":     2,
+			"int 0; int 0; app_params_get AppApprovalProgram": 5,
+			"byte 0x01; log":                                  5,
+			sender + "acct_params_get AcctBalance":            7,
```

Comment:
no boxes?

Reply:
Added
```
			"byte 0x1234; int 12; box_create":             8,
			"byte 0x1234; int 12; int 4; box_extract":     8,
			"byte 0x1234; int 12; byte 0x24; box_replace": 8,
			"byte 0x1234; box_del":                        8,
			"byte 0x1234; box_len":                        8,
			"byte 0x1234; box_get":                        8,
			"byte 0x1234; byte 0x12; box_put":             8,

```

I don't think this is a very powerful test, but why not?